### PR TITLE
build: add Requires.private to soletta.pc

### PR DESCRIPTION
**Pull request by dorileo**
This patch introduces a new private-requires-y and private-requires-m
variables to declare which pkg dependencies we're using, with that we
can add that to our distributed soletta.pc file.

For that we're also considering env vars in the template parser so we
can easily export&get useful information while processing template
files.

Signed-off-by: Leandro Dorileo <leandro.maciel.dorileo@intel.com>

### CODE REVIEW

File: pc/soletta.pc.in [hunk 1]
```diff
@@ -11,3 +11,4 @@ Description: SOLETTA io library
 Version: {{@VERSION@}}
 Libs: -L${libdir} -lsoletta
 Cflags: -I${includedir}/soletta
+Requires.private: {{@requires-private@}}
```
Comment (barbieri) [inline pc/soletta.pc.in]: missing trailing whitespace.

File: tools/build/Makefile.rules [hunk 2]
```diff
@@ -212,7 +212,9 @@ inc-subdirs = \
 			$(call parse-sample,$(sample),$(subdir)) \
 		) \
 		$(clean-control) \
-	)
+	) \
+	$(eval requires-private := $(sort $(requires-private-y) $(requires-private-m))) \
```
Comment (barbieri) [inline tools/build/Makefile.rules]: only `requires-private-y` is needed. Whatever goes as shared modules/objects doesn't count in libsoletta.so and thus not in `soletta.pc`

Comment (dorileo) [inline tools/build/Makefile.rules]: all right, I was unsure about that. Thanks.

File: pc/soletta.pc.in (hunk 1 above)
Comment (dorileo) [inline pc/soletta.pc.in]: ok

**Comment by dorileo** [comment]
v2, coming.

**Pull request closed without merging**